crates/router/src/routes/payments.rs

```diff
@@ -2881,9 +2881,17 @@ pub async fn payment_status(
     let global_payment_id = path.into_inner();
     tracing::Span::current().record("payment_id", global_payment_id.get_string_repr());
 
+    let payload = payment_types::PaymentsRetrieveRequest {
+        force_sync: payload.force_sync,
+        expand_attempts: payload.expand_attempts,
+        param: payload.param.clone(),
+        all_keys_required: payload.all_keys_required,
+        ..Default::default()
```

Comment:
If we use `..Default::default()`, we won't be notified by the compiler when we add new fields to the struct. Is that intentional, or would we prefer to explicitly pass new fields whenever they're introduced?